diff --git a/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs b/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs
index 7f64dc3df2..891027478d 100644
--- a/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs
+++ b/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs
@@ -590,7 +590,9 @@
         if let Some((bounding_span, source, offset_line)) = shrink_file(&spans, file_name, sm) {
             let adj_lo = bounding_span.lo().to_usize();
             let filename = sm.filename_for_diagnostics(file_name).to_string_lossy().to_string();
-            Some(Snippet::source(source).line_start(offset_line).path(filename).annotations(
+            let path_url = annotations.iter().find_map(|annotation| matches!(annotation.kind, AnnotationKind::Primary).then_some(annotation.span))
+                .and_then(|span| self.path_url(sm, file_name, span));
+            Some(Snippet::source(source).line_start(offset_line).path(filename).path_url(path_url).annotations(
                 annotations.into_iter().map(move |a| {
                     let lo = a.span.lo().to_usize().saturating_sub(adj_lo);
                     let hi = a.span.hi().to_usize().saturating_sub(adj_lo);
@@ -639,10 +641,12 @@
                     // ⸬  $SRC_DIR/core/src/option.rs:602:4
                     // │
                     // ╰ note: not covered
+                    let path_url = self.path_url(sm, file_name, a.span);
                     group = group.element(
                         Origin::path(filename.clone())
                             .line(sm.doctest_offset_line(file_name, lo.line))
-                            .char_column(lo.col_display),
+                            .char_column(lo.col_display)
+                            .path_url(path_url),
                     );
                 }
 
@@ -655,10 +659,12 @@
                     // ⸬  $SRC_DIR/core/src/option.rs:602:4 (<- It adds *this*)
                     // │
                     // ╰ note: not covered
+                    let path_url = self.path_url(sm, file_name, a.span);
                     group = group.element(
                         Origin::path(filename.clone())
                             .line(sm.doctest_offset_line(file_name, hi.line))
-                            .char_column(hi.col_display),
+                            .char_column(hi.col_display)
+                            .path_url(path_url),
                     );
                 }
 
@@ -677,6 +683,21 @@
         }
         group
     }
+
+    fn path_url(&self, sm: &SourceMap, file_name: &FileName, span: Span) -> Option<String> {
+        // if !matches!(self.terminal_url, TerminalUrl::Yes) {
+        //     dbg!();
+        //     return None;
+        // }
+        let hostname = nix::unistd::gethostname().ok()?;
+        let absolute_path = file_name.clone().into_local_path()?.canonicalize().ok()?;
+        let hostname = hostname.display();
+        // TODO(scrabsha): characters must be encoded here probably.
+        let absolute_path = absolute_path.display();
+        let line = sm.lookup_char_pos(span.lo()).line;
+        
+        Some(format!("file://{hostname}/{absolute_path}:{line}"))
+    }
 }
 
 fn emit_to_destination(
